src/postgres/contrib/yb_pg_metrics/yb_pg_metrics.c

```diff
@@ -528,6 +534,29 @@ ybpgm_ProcessUtility(PlannedStmt *pstmt, const char *queryString,
   {
     instr_time start;
     instr_time end;
+    statementType type;
+
+    if (IsA(pstmt->utilityStmt, TransactionStmt)) {
+      TransactionStmt *stmt = (TransactionStmt *)(pstmt->utilityStmt);
+      switch (stmt->kind) {
+      case TRANS_STMT_BEGIN:
+        type = Begin;
+        break;
+      case TRANS_STMT_COMMIT:
+        type = Commit;
+        break;
+      case TRANS_STMT_ROLLBACK:
+      case TRANS_STMT_ROLLBACK_TO:
+        type = Rollback;
+        break;
+      default:
```

Comment:
Handle all statement kinds, particularly `TRANS_STMT_START`.  I would shape the code so that there's a failure when not all cases are explicitly handled.  The rationale is that future additions to the enum should cause equal changes here.